Branch-SDK/src/io/branch/referral/BranchRemoteInterface.java

```diff
@@ -270,23 +272,26 @@ public void sendLog(String log) {
 		}
 	}
 	
-	private String convertJSONtoString(JSONObject json) {
+	private String convertJSONtoString(JSONObject json) throws JSONException {
 		StringBuilder result = new StringBuilder();
 		
 		if (json != null) {
-	        Iterator<String> iter = json.keys();
-	        if (iter.hasNext()) {
-	        	boolean first = true;
-	        	do {
+            JSONArray names = json.names();
+	        if (names != null) {
+                boolean first = true;
+                int size = names.length();
+                for(int i = 0; i < size; i++) {
+
+                    String key = names.getString(i);
+
 	        		if (first) {
 		        		result.append("?");
 		        		first = false;
 		        	} else {
 		        		result.append("&");
 		        	}
-	        		
-	        		String key = iter.next();
-	        		String value;
+
+                    String value;
 	        		try {
 						value = json.getString(key);
 					} catch (JSONException e) {
```

Comment:
With the method signature having "throws JSONException", we can remove the try/catch here?